### PR TITLE
add runstats recipe

### PR DESCRIPTION
**Pull request by gschramm**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by gschramm** [comment]
@conda-forge/help-python-c ready for review

**Comment by gschramm** [comment]
Anybody an idea why "pytest -k" works as intended on Linux/OSX but fails on Windows? I don't understand why it complains that it cannot find a file/dir "test_multiply". -k 'not test_multiply' is supposed to skip this test
```
2023-12-14T21:11:02.6552791Z (%PREFIX%) %SRC_DIR%>pytest tests/test_runstats.py -vs -k 'not test_multiply' 
2023-12-14T21:11:03.1170725Z ============================= test session starts =============================
2023-12-14T21:11:03.1177686Z platform win32 -- Python 3.12.0, pytest-7.4.3, pluggy-1.3.0 -- %PREFIX%\python.exe
2023-12-14T21:11:03.1180719Z cachedir: .pytest_cache
2023-12-14T21:11:03.1182191Z rootdir: %SRC_DIR%
2023-12-14T21:11:03.1186939Z collecting ... collected 0 items
2023-12-14T21:11:03.1231168Z 
2023-12-14T21:11:03.1235700Z ============================ no tests ran in 0.00s ============================
2023-12-14T21:11:03.1238095Z ERROR: file or directory not found: test_multiply'
```

File: recipes/runstats/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/grantjenks/python-runstats/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 48a0cd627ee5e854990960ca8920a789922fe7477d0c6d469f9f5baf6e007294
+
+build:
+  noarch: python
```
Comment (xhochy) [inline recipes/runstats/meta.yaml:13]: If you need a compiler, it cannot be `noarch`

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/runstats/meta.yaml (hunk 1 above)
Comment (gschramm) [reply recipes/runstats/meta.yaml]: Thanks for the review and good catch. I started from `grayskull pypi` which introduced the `noarch` (was not aware when starting that it uses `cython`).

The linter now says: `Non noarch packages should have python requirement without any version constraints` which is why I removed the python constrained in `host` and `run` but that does not feel correct to me.

**Review by xhochy — APPROVED** [review]

**Pull request merged**